### ISSUE TITLE
When .git is removed from Terminal, it isn't reflected in the sidebar Files Panel & Git Panel

### ISSUE THREAD

**Issue by deadcoder0904** [Microsoft/vscode#53520]
- VSCode Version: 1.24.1
- OS Version: Mac OS 10.13.5

Steps to Reproduce:

1. Clone a git repo
2. Edit some files to see some git diffs in Git Panel
3. Launch Terminal inside VSCode & then remove `.git` folder
4. You will still see Git diffs in side panel 

Does this issue occur when all extensions are disabled?: Yes

**Comment by joaomoreno** [comment]
Oh yeah... this has broken since a while...

**Comment by JacksonKearl** [comment]
Still seeing the diffs until I reload the window

**Comment by joaomoreno** [comment]
Created https://github.com/Microsoft/vscode/issues/55729

**Comment by joaomoreno** [comment]
@chrmarti I can't repro that, still works for me when there are no commits. Can you double check?

**Comment by chrmarti** [comment]
As a movie:
![53520](https://user-images.githubusercontent.com/9205389/43726749-440e2b86-99a0-11e8-901c-bb5bdbcf9290.gif)

**Comment by joaomoreno** [comment]
It doesn't react if you wait ~10 seconds?

**Comment by chrmarti** [comment]
No, although I have seen it update, but only 1 out of 4 times or so.

**Comment by joaomoreno** [comment]
Yup, I could reproduce what you see @chrmarti. Unfortunately, what happens is the git extension doesn't get file events, for which @bpasero probably already has an issue for. If the file events do reach the git extension, then the UI behaves correctly. Closing this.

**Comment by bpasero** [comment]
@chrmarti to verify that behaviour can you run "code --verbose" and see the output for the file watchers?

**Comment by joaomoreno** [comment]
@bpasero I ran it out of dev and `console.log` file events as they came from the watcher API. When the bug didn't repro, I was getting events. When the bug repros, no events were coming.

**Comment by chrmarti** [comment]
The output from `--verbose` from right before I initialize the repository to right after I `rm -rf .git` and waited for 10+ seconds: [53520.txt](https://github.com/Microsoft/vscode/files/2348804/53520.txt)

**Comment by bpasero** [comment]
@chrmarti you will have to copy it out of devtools, the terminal output misses all the info

**Comment by bpasero** [comment]
(I fixed that in https://github.com/Microsoft/vscode/commit/25558ca524e62b782014e5ef97438474cc32f42e so it should work now if running out of sources)

**Comment by chrmarti** [comment]
Here you go: [53520-console.log](https://github.com/Microsoft/vscode/files/2348984/53520-console.log)

**Comment by chrmarti** [comment]
Yes, not always (as before), but it repros. (Also just tried running out-of-source.)

**Comment by bpasero** [comment]
@joaomoreno I can reproduce:
* open folder without git repo
* initialize
* remove .git folder from integrated terminal

The extension host is seeing the file event, so it must be a bug in the extension:

![image](https://user-images.githubusercontent.com/900690/45079012-b11ee900-b0f1-11e8-8c87-36a51c5aaf0d.png)
